internal/config/pipeline.go

```diff
@@ -99,8 +104,48 @@ func (c *Config) Pipeline(id string, ctx *hcl.EvalContext) (*Pipeline, error) {
 			Use:       stepRaw.Use,
 		}
 
+		// Parse a nested pipeline step if defined
+		// TODO(briancain): At the moment, we're supporting singly nestested Pipeline
+		// stanzas. If we want to support fully nested embedded pipelines we'll need
+		// to turn this into a recursive function.
+		if stepRaw.PipelineRaw != nil {
+			var embedPipeline Pipeline
+			if diag := gohcl.DecodeBody(stepRaw.PipelineRaw.Body, finalizeContext(ctx), &embedPipeline); diag.HasErrors() {
+				return nil, diag
+			}
+
+			s.Pipeline = &Pipeline{
+				Name:   stepRaw.PipelineRaw.Name,
+				ctx:    ctx,
+				Config: c,
+			}
+			if s.Pipeline.Config != nil {
+				s.Pipeline.Config.ctx = ctx
+			}
+
+			// Parse all the steps
+			var embSteps []*Step
+			for _, embedStepRaw := range embedPipeline.StepRaw {
+				// TODO(briancain): Do we add an error here if we detect another level
```

Comment:
In case you're looking for an answer for this PR, then I think yes we should, so long as embedded pipelines won't work right now - which I think is the case until this function is recursive (as you mentioned in your earlier comment).

Reply:
Done!